lib/beaker-hostgenerator/generator.rb

```diff
@@ -104,6 +112,19 @@ def __parse_node_info_token(token)
         node_info['arbitrary_roles'] = []
       end
 
+      if node_info['host_settings']
```

Comment:
I would probably extract this code to a method with an intention-revealing name.